lib/improver/utilities/cube_units.py

```diff
@@ -0,0 +1,197 @@
+            This is a list of coordinate names.
+    Keyword Args:
+        inplace (bool):
+            If True (default) the cubes are modified in place, if False this
+            function returns a modified copy of the cubes.
+    Returns:
+        cubes (iris.cube.CubeList or insitu):
+            The input cubes with units and data types of the chosen coordinates
+            set to match the definitions in units.py.
+    Raises:
+        KeyError: If coordinate to enforce is not defined in units.py
+        ValueError: If requested unit conversion is not possible.
+        ValueError: If a unit data type could not be converted without losing
+                    significant information (e.g. rounding time to the nearest
+                    hour when there are sub-hourly components).
+    """
+    if not inplace:
+        cubes = [cube.copy() for cube in cubes]
+
+    for cube in cubes:
+        for coord_name in coordinates:
+            try:
+                unit = DEFAULT_UNITS[coord_name]["unit"]
+                dtype = DEFAULT_UNITS[coord_name]["dtype"]
+            except KeyError:
+                msg = "Coordinate {} not defined in units.py"
+                raise KeyError(msg.format(coord_name))
+
+            try:
+                coordinate = cube.coord(coord_name)
+                coordinate.convert_units(unit)
+            except ValueError:
+                msg = '{} units cannot be converted to "{}"'
+                raise ValueError(msg.format(coord_name, unit))
+            except CoordinateNotFoundError:
+                pass
+            else:
+                if check_precision_loss(dtype, coordinate.points):
+                    coordinate.points = coordinate.points.astype(dtype)
+                else:
+                    msg = ('Data type of coordinate "{}" could not be'
+                           ' enforced without losing significant precision.')
+                    raise ValueError(msg.format(coord_name))
+
+    if not inplace:
+        return cubes
+
+
+def enforce_diagnostic_units_and_dtypes(cubes, inplace=True,
+                                        check_precision=False):
+    """
+    Function to enforce diagnostic units and data types as defined in units.py.
+
+    By default the diagnostic units are changed in place, but setting inplace
+    to False will return a copy of the cubes, leaving the originals unchanged.
+
+    Args:
+        cubes (iris.cube.CubeList):
+            List of cubes on which to enforce diagnostic units and data types.
+    Keyword Args:
+        inplace (bool):
+            If True (default) the cubes are modified in place, if False this
+            function returns a modified copy of the cubes.
+        check_precision (bool):
+            If True the change of cube units and data types is checked to
+            ensure precision is not lost, e.g. losing decimal places when
+            converting the data type. This defaults to False.
```

Comment:
What's the logic behind this option?  I'd assume we'd want to check for precision loss automatically and raise an error, as you've coded for coordinate enforcement above.

Reply:
Again, I think you are right. I'll change this behaviour.

Reply:
I've made this mandatory and altered the tests to reflect this.